launch/launch/launch_service.py

```diff
@@ -43,6 +44,19 @@
 from .utilities import visit_all_entities_and_collect_futures
 
 _logger = logging.getLogger('launch.LaunchService')
+_g_loops_used = set()
+
+
+# This atexit handler ensures all the loops are closed at exit.
+# This is only really required pre-3.6, see:
+#   https://github.com/ros2/launch/issues/84
+#   https://bugs.python.org/issue23548
+@atexit.register
+def close_loop():
+    global _g_loops_used
+    for loop in _g_loops_used:
+        if not loop.is_closed():
+            loop.close()
```

Comment:
I would be good to double check the behavior on Windows since I have seen indefinite blocking when trying to close the loop.

Reply:
I'll check it manually, but there is at least one automated test that runs the launch service and then exits (though `pytest` might influence the behavior if it also uses `asyncio`).

Reply:
I'm not sure if this pr caused it (I don't think so), but I'm investigating an issue with `launch` on Windows anyways, so I'll make sure everything is working there: https://github.com/ros2/launch/issues/86